workerpool/pool.go

```diff
@@ -46,7 +48,11 @@ func (pool *WorkerPool) Start() {
 
 				switch event.Type {
 				case watcher.Created:
-					job, err := newJob(pool.session, event.Connection, pool.log.Named("job"))
+					if pool.numWorkers+event.Connection.Source.NumberOfWorkers() > pool.maxWorkers {
+						continue
```

Comment:
Should we log here just in case? Might be good to know we're hitting this before any of our alerting?